tests/common/src/pkcs11/CMakeLists.txt

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (C) 2023 Renesas Electronics Corporation.
+# Copyright (C) 2023 EPAM Systems, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET pkcs11_test)
+
+# ######################################################################################################################
+# Setup SoftHSM2
+# ######################################################################################################################
+
+find_library(
+    SOFTHSM2_LIB
+    NAMES softhsm2
+    PATH_SUFFIXES softhsm
+)
+
+if(NOT SOFTHSM2_LIB)
+    message(FATAL_ERROR "softhsm2 library not found")
+endif()
+
+set(SOFTHSM_BASE ${CMAKE_CURRENT_BINARY_DIR}/softhsm)
+
+file(MAKE_DIRECTORY ${SOFTHSM_BASE}/tokens/)
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/softhsm2.conf ${SOFTHSM_BASE}/softhsm2.conf COPYONLY)
+file(APPEND ${SOFTHSM_BASE}/softhsm2.conf "directories.tokendir = ${SOFTHSM_BASE}/tokens/\n")
+
+# ######################################################################################################################
+# Create PKCS11 test certificates
+# ######################################################################################################################
+
+set(CERTDIR ${CMAKE_CURRENT_BINARY_DIR}/certificates)
+
+file(MAKE_DIRECTORY ${CERTDIR})
+
+message("\nGenerating PKCS11 test certificates...")
+
+message("  1) Create a Certificate Authority private key...")
```

Comment:
remove numbering: 1), 2) - when you have to add or remove something you should always reorder this.